src/main/java/emissary/core/BaseDataObject.java

```diff
@@ -37,6 +38,11 @@
     /* Our payload */
     protected byte[] theData;
 
+    /**
+     * A string representation of the data
+     */
+    protected String dataString = null;
```

Comment:
Although we don't serialize this, it might be reasonable to mark this `transient` as a indication to a fellow developer that this is expected to be a 'cached' representation of the data as a string.